### PR TITLE
feat: json traversal

### PR DESCRIPTION
**Pull request by dmfay**
this needs more time than I have for it at the moment but I don't think it'll be too hard to get basic support for #166! element indexing (possibly some common ground with the `array[1,2,3]` syntax?) and jsonpath are bigger considerations though.

### CODE REVIEW

**Comment by matthias-Q** [comment]
I added the other json operators and updated the test. Actually it looks like this is it. All other json operations are probably handled by `invocation`

**Comment by Gelio** [comment]
Drive-by appreciation comment:

I am honestly amazed by your aptitude and the promptness at fixing the issues I've reported. Stellar project maintenance! I certainly wasn't expecting to see all 3 issues addressed the same day I reported them

**Comment by dmfay** [comment]
> Drive-by appreciation comment:
>
> I am honestly amazed by your aptitude and the promptness at fixing the issues I've reported. Stellar project maintenance! I certainly wasn't expecting to see all 3 issues addressed the same day I reported them

the JSON issue has secretly been bugging me for a while (along with array literals and subscripting, but that's for later) and I've always been too busy with other things & other projects to take a look, and it turns out it just fits right into the `expression` node now

**Review by matthias-Q — APPROVED** [review]
LGTM. Thanks for fixing the tests.

File: test/corpus/json.txt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   (select_expression
+    (term
+     (json_traversal
+      (literal))
+     (keyword_as)
+     (identifier))
+    (term
+     (json_traversal
+      (json_traversal
+       (literal))))
```
Comment (DerekStride) [inline test/corpus/json.txt:28]: This subtree represents the term `"user"->'address' ->> 'city'`. It only references a single `literal` node which means the other two have been hidden.

I've puts together https://github.com/DerekStride/tree-sitter-sql/pull/171 to show an alternative using the `binary_expression` node that will surface all 3 literals and treat the navigation nodes as anonymous operators.

**Comment by matthias-Q** [comment]
We should add the new operators to the highlights.scm (i forgot that with `filter` yesterday as well ...)

**Comment by matthias-Q** [comment]
This looks good to me. I am merging it. I am so surprised that this went so smoothly.

(I was hoping that after the recent refactor we could also tackle `array` and `not in`, but it seems like they are still difficult to solve)

**Comment by dmfay** [comment]
> This looks good to me. I am merging it. I am so surprised that this went so smoothly.
>
> (I was hoping that after the recent refactor we could also tackle `array` and `not in`, but it seems like they are still difficult to solve)

I went looking and we [actually got arrays](https://github.com/DerekStride/tree-sitter-sql/blob/main/test/corpus/select.txt#L2398) at some point!

**Comment by matthias-Q** [comment]
@dmfay yes, but not as a function. See #115 Besides, I think #128 is more important :-/

**Pull request merged**